File: tests/fsharpqa/Source/Conformance/ObjectOrientedTypeDefinitions/AbstractMembers/BaseClassAndInterface.cs
namespace CSLibrary
{
    public interface I
    {
        void M();
    }

    public class C : I
    {
        public virtual void M() { System.Console.WriteLine("I am my method"); }

        void I.M()
        {
            System.Console.WriteLine("I am going via the interface");
            M();
        }
    }

}

File: tests/fsharpqa/Source/Conformance/ObjectOrientedTypeDefinitions/AbstractMembers/BaseClassAndInterface.fs
// #Conformance #ObjectOrientedTypes #MethodsAndProperties #MemberDefinitions 
module FSLibrary

// Interface with a member
type I = abstract M : unit -> unit

type C = 
    new() = {}
    abstract member M : unit -> unit
    default this.M() = System.Console.WriteLine("I am my method")
    interface I with
        member this.M() = 
            System.Console.WriteLine("I am going via the interface")
            this.M()
